File: javascript/grains/grains.js
export const square = n => {
  let casilla = 1;
  let granos = 1n;
  if (n > 0 && n < 65) {
    for (casilla; casilla < n; casilla++) {
      granos *= 2n;
    }
  } else {
    throw new Error("square must be between 1 and 64");
  }

  return granos;
};

export const total = () => {
  let total = 0n;
  let grains = 1n;
  for (let i = 1; i < 65; i++) {
    grains *= 2n;
  }
  return (total += grains) - 1n;
};

File: ruby/grains/grains.rb
# frozen_string_literal: true

# Class Grains
#
class Grains
  MAX_BOARD = 64
  MIN_BOARD = 1
  BASE = 2

  def self.square(num)
    raise ArgumentError if num < MIN_BOARD || num > MAX_BOARD

    BASE**(num - MIN_BOARD)
  end

  def self.total
    (1..MAX_BOARD).reduce { |acc, value| acc + square(value) }
  end
end
